src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportResourceLoader.cs

```diff
@@ -67,30 +66,63 @@ private async Task LoadResourcesInternalAsync(Channel<ImportResource> outputChan
                 using var stream = _integrationDataStoreClient.DownloadResource(new Uri(resourceLocation), offset, cancellationToken);
                 using var reader = new StreamReader(stream);
 
-                string content = null;
                 long currentIndex = 0;
                 long currentBytesRead = 0;
                 var buffer = new List<(string content, long index, int length)>();
 
+                var endOfStream = false;
+                StringBuilder line = new StringBuilder();
+                int calcultedEndOfLineLength = 0;
                 var skipFirstLine = true;
+
 #pragma warning disable CA2016
-                while ((currentBytesRead <= bytesToRead) && !string.IsNullOrEmpty(content = await reader.ReadLineAsync()))
+                while ((currentBytesRead <= bytesToRead) && !endOfStream)
 #pragma warning restore CA2016
                 {
                     cancellationToken.ThrowIfCancellationRequested();
 
-                    if (offset > 0 && skipFirstLine) // skip first line
+                    line = new StringBuilder();
+                    char[] c = new char[1];
+                    while (await reader.ReadAsync(c, 0, c.Length) > 0)
                     {
-                        skipFirstLine = false;
+                        char currentChar = c[0];
+
+                        if (currentChar == '\n')
+                        {
+                            calcultedEndOfLineLength = 1;
+                            break;
+                        }
+                        else if (currentChar == '\r')
+                        {
+                            char nextChar = (char)reader.Peek();
+                            if (nextChar == '\n')
+                            {
+                                calcultedEndOfLineLength = 2;
+                                await reader.ReadAsync(c, 0, c.Length);
+                                break;
+                            }
+                        }
```

Comment:
For reference: https://github.com/ndjson/ndjson-spec/blob/master/README.md#32-parsing

Reply:
Also looks like something we should add some tests for :)